sd-validation/schema.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const walkObject = require('./helpers/walkObject');
+// @ts-ignore
+const schemaStructure = new Map(require('./assets/schema_google'));
+const TYPE_KEYWORD = '@type';
+const SCHEMA_ORG_URL = 'http://schema.org/';
```

Comment:
both http and https versions could be used in defining schema types so later in cleanName we risk to have an issue

Reply:
shall we just normalize all http -> https